src/applications/personalization/profile-2/tests/components/Profile.data-unavailable-error.unit.spec.js

```diff
@@ -93,6 +93,32 @@ async function errorAppearsOnAllPages(
   });
 }
 
+async function hideErrorNonVet(
+  pageNames = [
+    PROFILE_PATH_NAMES.MILITARY_INFORMATION,
+    PROFILE_PATH_NAMES.DIRECT_DEPOSIT,
+    PROFILE_PATH_NAMES.ACCOUNT_SECURITY,
+    PROFILE_PATH_NAMES.CONNECTED_APPLICATIONS,
+  ],
+) {
+  const initialState = createBasicInitialState();
+  initialState.user.profile.veteranStatus = null;
+
+  const view = render(<Profile2Router isLOA3 isInMVI />, {
+    initialState,
+  });
+  const spinner = view.queryByRole('progressbar');
+  await waitForElementToBeRemoved(spinner);
+
+  const alert = await view.queryByTestId(ALERT_ID);
+  expect(alert).not.to.exist;
+
+  pageNames.forEach(pageName => {
+    userEvent.click(view.getByRole('link', { name: pageName }));
+    expect(alert).to.be.null;
```

Comment:
I think that for this `expect` to do what you want you'd really need to rerun the query:
`expect(view.queryByTextId(ALERT_ID)).to.not.exist;`

Reply:
Made the update, thanks for the suggestion!